### ISSUE TITLE
Possibly add metrics for newly added Mongo4 connection pool events

### ISSUE THREAD

**Issue by bono007** [micrometer-metrics/micrometer#2499]
### Background
While migrating to Mongo client 4.x in #2493 several new connection pool events were noticed (see below). They were marked out of scope for the initial migration in favor of this follow up ticket. 
* [`connectionCheckOutStarted`](https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/event/ConnectionPoolListener.java#L71)
* [`connectionCheckOutFailed`](https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/event/ConnectionPoolListener.java#L88)
* [`connectionReady`](https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/event/ConnectionPoolListener.java#L126)
* [`connectionPoolCleared`](https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/event/ConnectionPoolListener.java#L54) (earlier [thread](https://github.com/micrometer-metrics/micrometer/pull/2493#discussion_r590961822) with question of the value of adding this one)

The MongoDB [Connection Monitoring and Pooling](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst) specification has details around each of the above events. 

### Acceptance Criteria
Each of the above connection pool events either has metrics added for it or reasoning around why metrics are not needed.

**Comment by codesimplicity** [comment]
Looks like this has been implemented some time ago 😉 

But I stumbled upon it when looking for an existing issue regarding [connectionCheckOutFailed](https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/event/ConnectionPoolListener.java#L88): this currently only decreases the `waitQueueSizes` gauge but does not record the failure in any way. I should probably raise a separate issue for that but any hint to similar handling would be appreciated.

**Comment by shakuzen** [comment]
There was recently [this issue](https://github.com/micrometer-metrics/micrometer/issues/4978) asking about exception/error tagging with MongoDB client metrics. Does that issue sound like it captures what you're looking for? I think we ideally would like users to tell us what would be useful rather than us introducing new things based on available events without knowing if it helps anyone.

**Comment by codesimplicity** [comment]
No, that's a different issue as it is about recording details when actually executing commands against the database.

I've now directly created a pull request: https://github.com/micrometer-metrics/micrometer/pull/5057
Not sure if that's the correct approach or if you prefer a separate issue 🤷

**Comment by codesimplicity** [comment]
@shakuzen coming back to my initial comment:

> Looks like this has been implemented some time ago 😉

I think this issue can be closed as [MongoMetricsConnectionPoolListener](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java) covers all of it. Or is there anything missing? (don't want to push, but looked like this issue got lost)
